pkg/smi/smi.go

```diff
@@ -0,0 +1,95 @@
+package smi
+
+import (
+	smiAccess "github.com/servicemeshinterface/smi-sdk-go/pkg/apis/access/v1alpha3"
+	smiSplit "github.com/servicemeshinterface/smi-sdk-go/pkg/apis/split/v1alpha2"
+)
+
+// FilterTrafficSplit applies the given TrafficSplitListOption filter on the given TrafficSplit object
+func FilterTrafficSplit(trafficSplit *smiSplit.TrafficSplit, options ...TrafficSplitListOption) *smiSplit.TrafficSplit {
```

Comment:
consider creating multiple methods for each List* call instead of having a generic filter.

Up to you